### PR TITLE
added two files for importnet.multinet

### PR DESCRIPTION
**Pull request by iyifan**

### CODE REVIEW

File: R/importNet.Multinet.R [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# importNet.MULTINET.R
+
+#' Import network data from BIOGRID, ENCODE data, KEGG, and SignaLink database 
+#' (Multinet interactions).
+#'
+#' \code{importNet.MULTINET} imports network edges from a MULTINET database file,
+#' 
+#' @section Networks:
+#'   MULTINET "protein.links.detailed.v10.txt" files contain
```
Comment (hyginn) [inline R/importNet.Multinet.R]: That's the file name and description of a STRING file.

File: R/importNet.Multinet.R [hunk 2]
```diff
@@ -0,0 +1,145 @@
+# importNet.MULTINET.R
```
Comment (RicardoRam) [inline R/importNet.Multinet.R]: Please include the R CMD check in a comment in pull request.

File: R/importNet.Multinet.R [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                             silent = FALSE,
+                             writeLog = TRUE) {
+
+    # ==== VALIDATIONS =========================================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fName,        like = "FILE_E",   checkSize = TRUE))
+    cR <- c(cR, .checkArgs(net,          like = "a",        checkSize = TRUE))
+    cR <- c(cR, .checkArgs(verbose,      like = logical(1), checkSize = TRUE))
```
Comment (RicardoRam) [inline R/importNet.Multinet.R]: check silent and writeLog?

File: tests/testthat/testImportNetMultinet.r [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    # correct log file
+    thisLog <- readLines(logName)
+    expect_true(grepl("event \\| title  \\| importNet.Multinet",   thisLog[1]))
+    expect_true(grepl("event \\| time",                            thisLog[2]))
+    
+    # call
+    expect_true(grepl("^event \\| call   \\| importNet.Multinet\\(", thisLog[3]))
+    expect_true(grepl(paste0("fname = \"", fN, "\""),                thisLog[3]))
+    expect_true(grepl("net = \"PPI\"",                               thisLog[3]))
+    expect_true(grepl("verbose = True",                              thisLog[3]))
```
Comment (RicardoRam) [inline tests/testthat/testImportNetMultinet.r]: I think it should be "verbose = TRUE"

File: R/importNet.Multinet.R [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+#'   allows users to define their own column.
+#' @section The gG object:
+#'   The function returns a rete gG object, a weighted,
+#'   directed, simple igraph graph in which HGNC gene symbols are vertex names,
+#'   and the edge attributes $weight hold the network scores. Graph attributes
+#'   hold metadata; use igraph::graph_attr(gG) to return: $version: the gG
+#'   object version; $logFile: the filename to which log information was
+#'   written; $inFile: the input filename of MULTINET data; $call: the complete
+#'   function call with expanded arguments; and $date: when the gG object was
+#'   created.
```
Comment (hyginn) [inline R/importNet.Multinet.R]: Add a note that weights in this case are all equal to 1.

File: R/importNet.Multinet.R [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+#'   Defaults to TRUE.
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @return a weighted, directed, simple igraph graph which is a rete gG object.
+#'
+#' @family ImportNet.STRING, importNet.MultiNet, importNet.MITAB
+#'
+#'   ## @seealso \code{\link{fastMap}} fastMap() is used internally to map ENSP
```
Comment (hyginn) [inline R/importNet.Multinet.R]: This is not relevant for MultiNet.

File: R/importNet.Multinet.R (hunk 3 above)
Comment (hyginn) [inline R/importNet.Multinet.R]: also check arguments for silent, writeLog ... !

File: R/importNet.Multinet.R [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+    if(length(cR) > 0) {
+        stop(cR)
+    }
+    
+    # Read header and one contents line
+    tmp <- readLines(fName, n = 2)
+    
+    data   <- unlist(strsplit(tmp[2], " "))
+    
+    # Parse for requested column. STRING data is " "-delimited.
```
Comment (hyginn) [inline R/importNet.Multinet.R]: This is not the right comment for MultiNet ...

File: R/importNet.Multinet.R [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    header <- unlist(strsplit(tmp[1], " "))
```
Comment (hyginn) [inline R/importNet.Multinet.R]: Is MultiNet delimeted by single blanks? If not, you have to change this line.

File: R/importNet.Multinet.R (hunk 3 above)
Comment (RicardoRam) [inline R/importNet.Multinet.R]: Hey Boris, I made the same comment earlier. You might need to refresh to see them.

File: R/importNet.Multinet.R [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+    iCol <- which(net == header)  # column that contains the requested net
+    
+    #  Validate that requested network exists in data
+    if (length(iCol) != 1) {
+        stop(paste0("Request error:", "\n",
+                    "Requested network type is ", sprintf("\"%s\"", net), "\n",
+                    "Found network type(s): <",
+                    paste(header[3:length(header)], collapse = ", "),
```
Comment (hyginn) [inline R/importNet.Multinet.R]: This too is not correct for the MultiNet format!

File: R/importNet.Multinet.R [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+                    ">", "\n"))
+    }
+
+    # ==== READ DATA ===========================================================
+
+    # Create column mask for readr::read_delim()
+    readMask <- paste("cc",
+                      paste(rep("_", length(header) - 2), collapse = ""),
```
Comment (hyginn) [inline R/importNet.Multinet.R]: This too is not correct for the MultiNet format.

File: R/importNet.Multinet.R [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+    # start reading
+    if (!silent) { cat("Reading", net, "from", fName, "...") }
+    
+    # Read file into data frame, use only selected column
+    netDF <- readr::read_delim(file = fName,
+                               delim = " ",
+                               col_types = readMask,
+                               n_max = Inf)
+    
+    colnames(netDF) <- c("a", "b", "weight")
```
Comment (hyginn) [inline R/importNet.Multinet.R]: IIRC MultiNet has both interactors in **one single string** in the first column, no? So you need to read this column, and process it to give you two interactor gene symbols.!

File: R/importNet.Multinet.R [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+    
+    nIn <- nrow(netDF)
+    if (!silent) { cat("done. (", nIn, " edges)", "\n") }
+    
+    
+    # ==== CREATE GRAPH OBJECT =================================================
+    
+    gG <- .df2gG(netDF)
```
Comment (hyginn) [inline R/importNet.Multinet.R]: gG is a directed graph. MultiNet has undirected edges. You need to do something about that. What?

File: R/importNet.Multinet.R [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+    
+    # ==== WRITE LOG ===========================================================
+    
+    if(writeLog) {
+      
+        myTitle <- "importNet.Multinet"
+      
+        # Compile function call record
+        myCall <- character()
+        myCall[1] <- "importNet.STRING("
```
Comment (hyginn) [inline R/importNet.Multinet.R]: This too is not correct for the MultiNet format.

File: R/importNet.Multinet.R [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+        myCall[2] <- sprintf("fname = \"%s\", ", fName)
+        myCall[3] <- sprintf("net = \"%s\", ", net)
+        myCall[4] <- sprintf("verbose = %s, ", as.character(verbose))
```
Comment (hyginn) [inline R/importNet.Multinet.R]: What is this line? It's not in your function signature at all!

File: R/importNet.Multinet.R [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+        myCall[5] <- sprintf("silent = %s, ", as.character(silent))
+        myCall[6] <- sprintf("writeLog = %s)", as.character(writeLog))
+        myCall <- paste0(myCall, collapse = "")
+      
+        # indicate input object name(s)
+        #   (NA for this importNet.Multinet())
+      
+        # Record progress information
+        myNotes <- character()
+        myNotes <- c(myNotes, sprintf("Read %s edges from file.", nIn))
```
Comment (hyginn) [inline R/importNet.Multinet.R]: Filename has to appear here!

File: tests/testthat/testImportNetMultinet.r [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+      gG <- importNet.Multinet(fName = fN,
+                               net = "noSuchNet",
+                               verbose = TRUE,
+                               silent = TRUE,
+                               writeLog = FALSE),
+      "Request error:")
+    
+    # ==== verbose
+    expect_error(
+      gG <- importNet.Multinet(fName = fN,
```
Comment (hyginn) [inline tests/testthat/testImportNetMultinet.r]: There is no argument verbose in your function!

File: R/importNet.Multinet.R [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+#'   hold metadata; use igraph::graph_attr(gG) to return: $version: the gG
+#'   object version; $logFile: the filename to which log information was
+#'   written; $inFile: the input filename of MULTINET data; $call: the complete
+#'   function call with expanded arguments; and $date: when the gG object was
+#'   created.
+#'
+#' @param fName Filename of a MultiNet Multinet.interactions.network_presence.txt file.
+#' @param net The requested network. This must be a MULTINET that exists in the
+#'   header. The default is "PPI".
+#' @param verbose Controls whether to print summary information after import.
```
Comment (hyginn) [inline R/importNet.Multinet.R]: I don't know where this is coming from - but there is no behaviour in the function that is controlled by this flag. Remove it!

File: R/importNet.Multinet.R [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+#' @family ImportNet.STRING, importNet.MultiNet, importNet.MITAB
+#'
+#'   ## @seealso \code{\link{fastMap}} fastMap() is used internally to map ENSP
+#'   IDs to gene symbols.
+#'
+#'   ## @examples ## \dontrun{ ## importNet.MULTINET(IN, OUT) ## }
+#' @export
+importNet.MultiNet <- function(fName,
+                             net = "PPI",
+                             verbose = TRUE,
```
Comment (hyginn) [inline R/importNet.Multinet.R]: remove "verbose"

File: R/importNet.Multinet.R (hunk 3 above)
Comment (hyginn) [inline R/importNet.Multinet.R]: remove "verbose"

File: tests/testthat/testImportNetMultinet.r [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
+# This is the unit test for importNet.Multinet()
+
+context("import networks")
+
+# ==== BEGIN SETUP AND PREPARE =================================================
+OLOG <- as.character(getOption("rete.logfile"))   # save original logfile name
+logFileName(fPath = tempdir(), setOption = TRUE)  # make tempdir() the log dir
+NL <- .PlatformLineBreak()
+# ==== END SETUP AND PREPARE ===================================================
+
```
Comment (hyginn) [inline tests/testthat/testImportNetMultinet.r]: All the "verbose" arguments need to be removed.

File: tests/testthat/testImportNetMultinet.r [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
+      '.checkArgs> "writeLog" mode error')
+    
+    # ==== no logfile should have been written
+    expect_false(file.exists(logName))
+    
+})
+
+
+test_that("importNet.Multinet() produces gG from Multinet data", {
+    fN <- "Multinet.interactions.network_presence.txt"
```
Comment (hyginn) [inline tests/testthat/testImportNetMultinet.r]: Where is this file located? You need a small(!) excerpt as test-data, not the whole thing! The test-data can be stored in this directory, or in ```inst/extdata/```. Check other functions' use of test data files for examples.

File: tests/testthat/testImportNetMultinet.r [hunk 21]
```diff
@@ -0,0 +1,200 @@
...
+    gG <- importNet.Multinet(fName = fN,
+                             net = "PPI",
+                             verbose = TRUE,
+                             silent = TRUE,
+                             writeLog = TRUE)
+    expect_equal(igraph::vcount(gG), 5)         # correct number of vertices
```
Comment (hyginn) [inline tests/testthat/testImportNetMultinet.r]: These numbers will depend on your actual test-data of course...

File: tests/testthat/testImportNetMultinet.r [hunk 22]
```diff
@@ -0,0 +1,200 @@
...
+})
+
+
+test_that("importNet.Multinet() doesn't produce gG from corrupt data", {
+    fN <- "Multinet.interactions.network_presence.txt"
+    
+    # ==== stop on missing NL on last line
+    inputdata <- file(fN, open = "r")
+    expect_error(
+      tail(inputdata, 1) <- !c("\n"),
```
Comment (hyginn) [inline tests/testthat/testImportNetMultinet.r]: ??? That is not one of the .checkArgs functions. Maybe not a bad idea - but right now this does not exist.

File: tests/testthat/testImportNetMultinet.r [hunk 23]
```diff
@@ -0,0 +1,200 @@
...
+      '.checkArgs> stop on missing newline on last line')
+    
+    # ==== stop on NA in requested network
+    network <- list("PPI", "METABOLIC", "GENETIC", "PHOSPHORYLATION",
+                    "REGULATORY", "SIGNALING", "NUM_NETWORKS")
+    expect_error(
+      NotAvailable <- !("net" %in% names(network)),  # "net" input in importNet.Multinet()
+      '.checkArgs> stop on not available in requested network')
```
Comment (hyginn) [inline tests/testthat/testImportNetMultinet.r]: This is not the error message that you would actually get.

File: tests/testthat/testImportNetMultinet.r [hunk 24]
```diff
@@ -0,0 +1,200 @@
...
+    
+    # ==== stop on mismatch between header elements and data columns
+    inputdata <- file(fN, open = "r")
```
Comment (hyginn) [inline tests/testthat/testImportNetMultinet.r]: I have no idea what you are doing here, and where you got this code from. Please explain.

File: tests/testthat/testImportNetMultinet.r [hunk 25]
```diff
@@ -0,0 +1,200 @@
...
+context("import networks")
+
+# ==== BEGIN SETUP AND PREPARE =================================================
+OLOG <- as.character(getOption("rete.logfile"))   # save original logfile name
+logFileName(fPath = tempdir(), setOption = TRUE)  # make tempdir() the log dir
+NL <- .PlatformLineBreak()
+# ==== END SETUP AND PREPARE ===================================================
+
+
+# ==== importNet.Multinet() ======================================================
```
Comment (RicardoRam) [inline tests/testthat/testImportNetMultinet.r]: Add comments for test cases.

**Comment by hyginn** [comment]
Code not complete, comments not addressed. Will  not be processed further.

**Pull request closed without merging**